src/audio/smart_amp/smart_amp_maxim_dsm.c

```diff
@@ -0,0 +1,593 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2020 Maxim Integrated All rights reserved.
+//
+// Author: Ryan Lee <ryans.lee@maximintegrated.com>
+
+#include <sof/audio/component.h>
+#include <sof/trace/trace.h>
+#include <sof/drivers/ipc.h>
+#include <sof/ut.h>
+
+#include <sof/audio/channel_map.h>
+#include <sof/trace/trace.h>
+#include <user/trace.h>
+#include <sof/bit.h>
+#include <sof/common.h>
+#include <ipc/channel_map.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <sof/audio/smart_amp/smart_amp.h>
+#include "dsm_api_public.h"
+
+static int maxim_dsm_init(struct smart_amp_mod_struct_t *hspk, struct comp_dev *dev)
+{
+	int *circularbuffersize = hspk->circularbuffersize;
+	int *delayedsamples = hspk->delayedsamples;
+	struct dsm_api_init_ext_t initparam;
+	enum DSM_API_MESSAGE retcode;
+
+	comp_dbg(dev, BUILD_TIME);
+
+	initparam.isamplebitwidth       = 16;
+	initparam.ichannels             = 2;
+	initparam.ipcircbuffersizebytes = circularbuffersize;
+	initparam.ipdelayedsamples      = delayedsamples;
+	initparam.isamplingrate         = 48000;
+
+	retcode = dsm_api_init(hspk->dsmhandle, &initparam,
+			       sizeof(struct dsm_api_init_ext_t));
+	if (retcode != DSM_API_OK) {
+		goto exit;
+	} else	{
+		hspk->ff_fr_sz_samples =
+			initparam.off_framesizesamples;
+		hspk->fb_fr_sz_samples =
+			initparam.ofb_framesizesamples;
+		hspk->channelmask = 0;
+		hspk->nchannels = initparam.ichannels;
+		hspk->ifsamples = hspk->ff_fr_sz_samples
+			* initparam.ichannels;
+		hspk->ibsamples = hspk->fb_fr_sz_samples
+			* initparam.ichannels;
+	}
+
+	comp_dbg(dev, "[DSM] Initialization completed. (module:%p, dsm:%p)",
+		 (uintptr_t)hspk,
+		 (uintptr_t)hspk->dsmhandle);
+
+	return 0;
+exit:
+	comp_err(dev, "[DSM] Initialization failed. ret:%d", retcode);
+	return (int)retcode;
+}
+
+#ifdef PRINT_ADAPTIVE_LOG
+static int maxim_dsm_get_adaptive_apram(struct smart_amp_mod_struct_t *hspk,
+					struct comp_dev *dev)
+{
+	enum DSM_API_MESSAGE retcode;
+	int cmdblock[1 + SMART_AMP_FF_MAX_CH_NUM];
+	int excur[2], temp[2], dc_res[2], fc[2], q[2];
```

Comment:
Any comment on what these value mean for debugging ?

Reply:
Shall remove this function because it was already commented out and same work can be done over mixer control.

Reply:
It's up to you, if it's useful for your debug or integration processes then keep it. It could be kept as a Kconfig option if needed by non developers to dump the data.